### DISCUSSION TITLE
file rename in templater replaces obsidian file explorer window

### DISCUSSION THREAD

**Discussion by wcerveny** [SilentVoid13/Templater#670] [Help] [answered]
I finally got the "file rename" functionality  (using "create new note from template") in my template to behave the way I wanted it to. However, when the commands in the template finish, the new markdown file replaces the obsidian file explorer window.  I can reopen the file explorer window via the command palette, but I'd prefer not to have to do this. How does one fix this so the new note doesn't replace the file explorer window, or is this some sort of a templater bug?

**Comment by welpdx** [comment, marked as answer]
Could you please paste your code here so I can better understand your problem? 

> the new markdown file replaces the obsidian file explorer window

When you say "obsidian file explorer window", are you referring to the left sidebar? If you are, here's a piece of code that works for me using a simple test vault when running `Templater: Create new note from template` Alt+N. 

```js
<%*
if (this.app.workspace.activeLeaf.getViewState()?.type =="file-explorer") {
this.app.workspace.setActiveLeaf(this.app.workspace.getMostRecentLeaf())
}
%>
```
Edit: added demo
<details>
           <summary style="font-size:20">Demo</summary>
           <img src="https://i.imgur.com/jJxLeIR.gif" alt="demo" />
</details>

**Reply by wcerveny** [reply]
Woohoo, this works!

Yes, I was referring to the left sidebar. Adding the code you recommended fixed the problem. I found the recommended code didn't work if it appears before the templater code, but it did work if I added it after the last part of the templater code.

For reference, my partially anonymized code with the recommended code at the end looks like:

~~~
---
<%*
    let item1 =  await tp.system.prompt("item1 number(s)")
    let Product = await tp.system.suggester(["product1", "product2", "product3", "product4", "product5"],["product1", "product2", "product3", "product4", "product5"],false,"Product Type")
    let Release = await tp.system.prompt(" Release")
    let Filename = "Item " + item1 + " " + Product + " " + Release
    await tp.file.rename("Item1 " + Filename) 
-%>
Title:  <% Filename %>
Item1:  <% item1 %>
Product:  <% Product %>
Release:  <% Release %>
Item URL: <% await tp.system.prompt("Item URL") %>
Item Reference: <% await tp.system.prompt("Item Reference") %>
Customers: <% await tp.system.prompt("Customer(s):") %>
Response needed:  <% await tp.system.suggester(["Yes","No"],["Yes","No"], false, "Response needed?") %>
---
<%*
if (this.app.workspace.activeLeaf.getViewState()?.type =="file-explorer") { this.app.workspace.setActiveLeaf(this.app.workspace.getMostRecentLeaf())
}
%>
~~~

Tnanks!

**Reply by welpdx** [reply]
yay!

**Comment by AB1908** [comment]
It is indeed a bug and #671 is related.

**Reply by welpdx** [reply]
This is the first post about this issue. #670 #671 #691
